eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Serialisation/EraIndepGenerators.hs

```diff
@@ -194,29 +140,10 @@ instance Mock c => Arbitrary (BHeader c) where
         bodySize
         bodyHash
 
-instance DSIGNAlgorithm c => Arbitrary (SignedDSIGN c a) where
-  arbitrary =
-    SignedDSIGN . fromJust . rawDeserialiseSigDSIGN
-      <$> (genByteString . fromIntegral $ sizeSigDSIGN (Proxy @c))
-
-instance CC.Crypto c => Arbitrary (BootstrapWitness c) where
-  arbitrary = do
-    key <- arbitrary
-    sig <- genSignature
-    chainCode <- ChainCode <$> arbitrary
-    attributes <- arbitrary
-    pure $ BootstrapWitness key sig chainCode attributes
-
-instance CC.Crypto c => Arbitrary (TP.HashHeader c) where
-  arbitrary = TP.HashHeader <$> genHash
-
-instance (Typeable kr, CC.Crypto c) => Arbitrary (WitVKey kr c) where
-  arbitrary =
-    WitVKey
-      <$> arbitrary
-      <*> arbitrary
+instance Crypto c => Arbitrary (TP.HashHeader c) where
+  arbitrary = TP.HashHeader <$> arbitrary
 
-instance CC.Crypto c => Arbitrary (Wdrl c) where
+instance Crypto c => Arbitrary (Wdrl c) where
   arbitrary = genericArbitraryU
   shrink = genericShrink
 
```

Comment:
I always wonder why Wdrl defined in  Cardano.Ledger.Shelley.Core
newtype Wdrl c = Wdrl {unWdrl :: Map (RewardAcnt c) Coin}
  deriving (Show, Eq, Generic)
  deriving newtype (NoThunks, NFData)

Isn't in the Core?  It probably should be introduced in  Cardano.Ledger.Address where RewardAcnt  is defined.

Reply:
Yeah, this is a really good point. By moving `Wdrl` to `Cardano.Ledger.Address` we would also be able to move `wdrlsTxBodyL` from `ShelleyEraTxBody` to `EraTxBody`.

I suggest this is done in a separate PR though. Cretaetd a ticket https://github.com/input-output-hk/cardano-ledger/issues/3194